api/controller/pin.js

```diff
@@ -1,88 +1,80 @@
-import { createPin } from "../model";
+import { createPin, deletePin, getPinById, updatePin } from "../model";
 import User from "../model/user";
 
 const addNewPin = async (req, res) => {
-  const { body = {} } = req;
+  const { body } = req;
+
+  if (!body.latitude || !body.longitude || !body.title) {
+    return res.status(400).json({ message: "Latitude, longitude, and title are required." });
+  }
```

Comment:
Can we do this in middleware using express-validator ???

https://express-validator.github.io/docs/

Reply:
I'll check

Reply:
@Mustafiz04 Well we will need to create a separate middleware for this thing and that will be some extra lines of code if we use express-validator and a extra dependency of the project. I think since express-validator has not much use for now in the project so this is fine enough.